s3-shared/guest-iam.tf

```
@@ -0,0 +1,123 @@
+resource "aws_iam_user" "guest" {
+  name = "guest-${var.guest_account_name}"
+  path = "/guest/"
+
+  tags = {
+    Project     = var.project
+    Environment = var.environment
+  }
+}
+
+data "aws_iam_policy_document" "s3" {
+  statement {
+    effect = "Allow"
+    actions = [
+      "s3:GetBucketAcl",
+      "s3:GetBucketLocation",
+      "s3:GetBucketVersioning",
+      "s3:GetObject",
+      "s3:GetObjectAcl",
+      "s3:GetObjectTagging",
+      "s3:GetObjectVersion",
+      "s3:ListBucket",
+      "s3:ListBucketVersions",
+      "s3:PutBucketVersioning",
+      "s3:PutObject",
+      "s3:PutObjectAcl",
+      "s3:PutObjectTagging",
+      "s3:DeleteObject",
+      "s3:DeleteObjectVersion"
```

```suggestion
      "s3:GetBucketAcl",
      "s3:GetBucketLocation",
      "s3:GetObject",
      "s3:GetObjectAcl",
      "s3:GetObjectTagging",
      "s3:GetObjectVersion",
      "s3:ListBucket",
      "s3:PutObject",
      "s3:PutObjectAcl",
      "s3:PutObjectTagging",
      "s3:DeleteObject"
```

* we should not allow guests to change the bucket versioning setting
* we should disallow guests to view/change versions of files; this allows us to see if they deleted anything